src/hooks/useAccountId.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState, useCallback, SetStateAction } from "react";
+import { AccountId } from "@hashgraph/sdk";
+import { hashConnectWallet } from "@/services/wallets/hashconnect/hashconnectClient";
+
+type FormValues = { [key: string]: string };
+
+export function useAccountId<T>(
+  setValues: SetStateAction<any>,
+  values: T,
+  columnName: string,
+) {
+  const [hederaAccountIdError, setHederaAccountIdError] = useState(false);
+
+  const handleAddressValueChange = useCallback(async () => {
+    const address = (values as FormValues)[columnName];
+
+    if (address && /([0-9][.][0-9][.][0-9])/.test(address)) {
+      try {
+        const evmAddress = await hashConnectWallet.getEvmAccountAddress(
```

Comment:
The hook will be usable across the app for any supported wallet, maybe it makes sense to move the `getEvmAccountAddress` method out of the `hashConnectWallet`? Considering it's not tied to the wallet itself, but its a just a rpc call, wdyt?